corehq/apps/cloudcare/views.py

```diff
@@ -111,6 +111,9 @@ def fetch_app(self, domain, app_id):
         return _fetch_build(domain, self.request.couch_user.username, app_id)
 
     def get_web_apps_available_to_user(self, domain, user):
+        if user['doc_type'] == 'WebUser' and not user.can_access_web_apps(domain):
```

Comment:
@esoergel It seemed ok to short circuit here and not do it in `get_application_access_for_domain`. What are your thoughts?

Reply:
FWIW, I'm working on https://dimagi.atlassian.net/browse/USH-4289 which is going to overlap with this, and having this check here is definitely better than having it intertwined with `ApplicationAccess`.